### DISCUSSION TITLE
What version of Node should be default?

### DISCUSSION THREAD

**Discussion by talesh** [wardenenv/warden#573] [Polls]
Node 10 is now sorely out of date and unsupported + dev teams will probably be requiring a new version of Node for development. 

- Should this be something configurable on .env?
- Would doing so spiral into too many images to support?

**Comment by sprankhub** [comment]
The node version is already configurable?

https://github.com/wardenenv/warden/blob/229f9c9fbd0119764efeaa0d75e2216075a575ed/commands/env-init.cmd#L83

And it seems as it installs pretty much any version you want:

https://github.com/wardenenv/images/blob/6d019b004cd42c0c60bae38141c96ef0e9f4028b/php-fpm/context/docker-entrypoint#L29-L35

**Reply by talesh** [reply]
Oh damn I was looking at the build image and assuming it was stuck on Node 10. 

https://hub.docker.com/layers/wardenenv/php-fpm/8.2-magento2/images/sha256-37150a9acca4e931a3ec3cd95e2f0106659e53e68d350b24a60a805fadc81a06?context=explore

**Comment by bap14** [comment]
To be fair, I think that the default environment initialization scripts use an older version of Node (see https://github.com/wardenenv/warden/blob/a90dbd09746000f4d0330c3f361063602c00abd2/commands/env-init.cmd#LL56C7-L56C18). While it's easy to customize, I think it would be worthwhile to determine which release of Node we should initially configure environments with (e.g. LTS version) and then allow the user to customize from there. I can't think of a reason to default to node 12 when 18 is the current LTS version.
